doc: OpenTitan: add 2020-02-21 notes (#1622)
tock/tock: doc/wg/opentitan-notes/2020-02-21.md

# Tock OT Notes 2/21/2020

Present:
- Laura Abbott, Oxide Computer
- Brad Campbell (Chair),  University of Virginia
- Jon Flatley, Google
- Alistair Francis, Western Digital
- Garret Kelly, Google
- Patrick Mooney, Oxide Computer
- Bryan Cantrill, Oxide Computer
- Adam Leventhal, Oxide Computer

## TODO

> - @gkelly : Open PR to add Silvestrs to the WG

Reviewer: ```suggestion
- @gkelly: Open PR to add Silvestrs to the WG
```

> - @jon-flatley: Open PR to add Silvestrs to the WG

Reviewer: ```suggestion
- @gkelly : Open PR to add Silvestrs to the WG
```

> - @jon-flatley: Investigate options for LLVM development

Reviewer: ```suggestion
- @gkelly: Investigate options for LLVM development
```

## libtock-rs
- RISC-V support not well maintained
  - Scripts not correct
  - Very difficult to build PMP test
  - Alistair: works if you build manually
- No CI support currently
  - Something we should work towards
  - Can use QEMU to help
- Goal: converge on a build system to make building libtock-rs easy on RISC-V to
  help ensure it gets more testing as we work towards CI
- Google: expect libtock-c to be supported on OT, but want top notch support for
  libtock-rs

## libtock-rs ownership
- Not clear that libtock-rs should be completely under the OT WG umbrella
- However, there is a lot of overlap between OT WG and libtock-rs development
  - Decided to see if we could add RISC-V support in libtock-rs under the OT WG
    umbrella in the same way that RISC-V support for the kernel is.

## Relocation Support
- Bryan: not having relocation for RISC-V/libtock-rs is an "iceberg" standing in
  the way
- Alistair: Difficult to have to change how apps compile when kernel size changes
  - Maybe could automate the process with some tooling?
- Alistair: Western Digital is working on GCC PIC support for RISC-V
  - Currently in the design phase
  - Working on an ABI

> - Jon: Need a better scope of work for what updates LLVM needs

Reviewer: ```suggestion
- Garret: Need a better scope of work for what updates LLVM needs
```

  - Can look for person/team at Google who could help
- Brad: How feasible is updating LLVM to do the correct relocations?
  - Laura: It seems like it is doable by an experience LLVM engineer

> - Jon: Google intends to use monolithic signed binaries initially to workaround

Reviewer: ```suggestion
- Garret: Google intends to use monolithic signed binaries initially to workaround
```
